### ISSUE TITLE
Lower-bound on zoom is 33%, labeling and documentation should reflect that

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#781]
While checking the fix on issue #778, I was playing around with the zoom levels.  I discovered that the lower-bound on the zoom is 33%.  If you move the slider or decrement the number in the drop-down menu, the graph doesn't get any smaller after that.

This is fine, it doesn't need to get any smaller, probably.  However, for consistency we should do the following
- [ ]  Our labeling needs to reflect this so that the zoom range is stated as 33 - 200% both in the menu and on the slider.
- [ ] The numbers in the drop-down menu should not go below 33%.
- [ ] The slider range should be adjusted so that 33% is the value when the slider is all the way to the left.
- [ ] The description of the zoom feature on the documentation page will need to reflect this as well.

I think we don't need to deal with this for the next release, we can fix it later.

**Comment by dondi** [comment]
So I dug into this and found that the zoom calculation algorithm is actually quite involved. Posting it here both as documentation and as a basis for deciding final behavior:
- The displayed percent range of [1% to 200%] actually maps to the scale factor [0.25, 4]
- However, the mapping isn't linear:
    * If the value is 100% or less, each percentage point is worth 0.0075
    * If the value is greater than 100%, each percentage point is worth 0.03
- Thus, a zoom input value of 75% maps to 0.8125 (i.e., 1 - ((100 - 75) * 0.0075))
- A zoom value of 175% maps to 3.25 (i.e., 1 + (75 * 0.03))
- Finally, whatever the scaled value is, it is clamped at 0.5 (i.e., as observed, % values of 33% or less make no difference)

Phew! That was tricky to even write down. I no longer recall the reasoning for this dual scale, but I will guess that it was meant to make the scale perception feel more visually intuitive.

The observed behavior in this issue is connected directly to the last bullet, which forces the final scale value to a minimum of 0.5. Based on the given linear mapping, a "1%" will actually go to a scale factor of 0.25.

I can release a beta which removes this clamping, and we can decide how this looks. That way, at least, the graph won't feel like it gets "stuck" below 33%; it will still scale down a little more, but it won’t disappear.

**Comment by kdahlquist** [comment]
I don't know of any reasoning. Nicole did this pretty autonomously.



Kam D. Dahlquist, Ph.D.
Professor of Biology
Loyola Marymount University

-------- Original message --------

**Comment by kdahlquist** [comment]
I played with the zoom.  It looks fine to me.  One issue with the zoom field in the View menu, though.  It will let me type in values that are outside the allowed range.  When I do this to other fields, it reverts the value I typed to something within the range (either the upper or lower bound, depending on what I typed).  I think it should behave the same here.

**Comment by dondi** [comment]
Oops, I thought I had dealt with it but clearly not. There was an issue while working on this where the range limitations were too aggressive: e.g., it was (for a time) impossible to directly type, say, "200" because upon typing "2," the range check jumped in and forced it to "25." I had to add some code to prevent that from happening but it looks like I broke any range checking at all. I’ll look into it.
